### DISCUSSION TITLE
How to use environment variable in react-static?

### DISCUSSION THREAD

**Discussion by paulnguyen-mn** [react-static/react-static#1465] [General] [answered]
Hi everyone, 
I'm looking for a solution to set environment variables for:
- Different `site root`: `dev`, `staging` and `production`.
- Different Google Analytics `tracking code`.
- And some more.
But I can't find a way to achieve so.

I've tried to create `.env` and play around with it but it doesn't work.
Can someone help? 
Thanks in advance 🙂

**Comment by paulnguyen-mn** [comment, marked as answer]
Yay, finally I found the answer.
I need to use `dotenv` package to achieve it.
Here is my steps:

1. Install `dotenv` package:

```
npm i --save-dev dotenv
```

2. Add `.env` file

```
GTAG_ID=UA-123124121-1
```

3. Load environment variables in `static.config.js`

```
require('dotenv').config()
console.log('ENVIRONMENT: ', process.env.GTAG_ID);
```

Done, now I can access to environment variables in static.config.js 😉

**Reply by sometimescasey** [reply]
Thank you!! I guess the normal `REACT_APP_` prefixed environment variables aren't picked up at the moment when this file gets read. This works!

**Reply by pappasG** [reply]
How can you separate the env variables for different environments?
